compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -908,7 +909,17 @@ trait Applications extends Compatibility {
       def simpleApply(fun1: Tree, proto: FunProto)(using Context): Tree =
         methPart(fun1).tpe match {
           case funRef: TermRef =>
-            val app = ApplyTo(tree, fun1, funRef, proto, pt)
+            var app = ApplyTo(tree, fun1, funRef, proto, pt)
+            if ctx.mode.is(Mode.UnsafeJavaReturn) then
```

Comment:
How do the modes UnsafeNulls and UnsafeJavaReturn interact? Specifically, what happens when both UnsafeNulls and UnsafeJavaReturn are enabled? Should UnsafeJavaReturn be a no-op within UnsafeNulls?

Reply:
I think `UnsafeNulls` and `UnsafeJavaReturn` should be two independent features. `UnsafeNulls` changes the type hierarchy for `Null`; `UnsafeJavaReturn` modifies the Java return types. They don't interfere each other. I think `UnsafeJavaReturn` shouldn't be a no-op in `UnsafeNulls`, because it can introduce non-nullable types to outside directly.